packages/backend/ubuntu-pro.go

```diff
@@ -0,0 +1,163 @@
+package main
+import (
+    "os/exec"
+    "context"
+    "errors"
+    pb "github.com/canonical/desktop-security-center/packages/proto"
+    "google.golang.org/protobuf/types/known/emptypb"
+    "github.com/tidwall/gjson"
+    "github.com/godbus/dbus/v5"
+)
+
+var conn *dbus.Conn
+var reqId string
+
+func isServiceEnabled(basename string) (bool, error) {
+    obj := conn.Object(
+        "com.canonical.UbuntuAdvantage",
+        dbus.ObjectPath("/com/canonical/UbuntuAdvantage/Services/" + basename),
+    )
+    status, err := obj.GetProperty("com.canonical.UbuntuAdvantage.Service.Status")
+    if err != nil {
+        return false, err
+    }
+    return status.Value().(string) == "enabled", nil
+}
+
+func (s *ProServer) IsMachineProAttached(ctx context.Context, _ *emptypb.Empty) (*pb.Boolean, error) {
+    obj := conn.Object(
+        "com.canonical.UbuntuAdvantage",
+        "/com/canonical/UbuntuAdvantage/Manager",
+    )
+    isAttached, err := obj.GetProperty("com.canonical.UbuntuAdvantage.Manager.Attached")
+    if err != nil {
+        return nil, err
+    }
+
+    return &pb.Boolean{ Ret: isAttached.Value().(bool) }, nil
+}
+
+func (s *ProServer) IsEsmInfraEnabled(ctx context.Context, _ *emptypb.Empty) (*pb.Boolean, error) {
+    enabled, err := isServiceEnabled("esm_2dinfra")
+    return &pb.Boolean{ Ret: enabled }, err
+}
+
+func (s *ProServer) IsEsmAppsEnabled(ctx context.Context, _ *emptypb.Empty) (*pb.Boolean, error) {
+    enabled, err := isServiceEnabled("esm_2dapps")
+    return &pb.Boolean{ Ret: enabled }, err
+}
+
+func (s *ProServer) IsKernelLivePatchEnabled(ctx context.Context, _ *emptypb.Empty) (*pb.Boolean, error) {
+    enabled, err := isServiceEnabled("livepatch")
+    return &pb.Boolean{ Ret: enabled }, err
+}
+
+var dbusServices map[dbus.ObjectPath]map[string]map[string]dbus.Variant
+func connectToSystemBus() error {
```

Comment:
Currently only the `ubuntu-pro` service makes use of `dbus`, but if you add any services in the future that also make use of `dbus`, I would move this call to connect to the system bus into the server.go file, get the connection once, and then pass it into each of the services.

Best way to do something like that is to define a new function on each service that takes in the `dbus conn` and returns a service. Something like this:
[Dbus connection obtained once](https://github.com/matthew-hagemann/ubuntu-desktop-provision/blob/main/provd/internal/services/manager.go#L30-L33)
[New function takes it in and makes it accessible to the service](https://github.com/matthew-hagemann/ubuntu-desktop-provision/blob/main/provd/internal/services/user/user.go#L44-L76)

Not immediately applicable but something to be mindful of :smile: